addons/account_edi_ubl_cii/models/account_edi_xml_ubl_20.py

```diff
@@ -596,6 +599,36 @@ def _import_fill_invoice_line_form(self, journal, tree, invoice, invoice_line, q
                 tax_nodes += elem.findall('.//{*}TaxSubtotal/{*}Percent')
         return self._import_fill_invoice_line_taxes(journal, tax_nodes, invoice_line, inv_line_vals, logs)
 
+    def _correct_invoice_tax_amount(self, tree, invoice):
+        """ The tax total may have been modified for rounding purpose, if so we should use the imported tax and not
+         the computed one """
+        if not invoice.is_invoice(include_receipts=True):
+            return
+
+        # For each tax in our tax total, get the amount as well as the total in the xml.
+        all_taxes = self.env['account.tax'].search([
+            ('company_id', '=', invoice.company_id.id),
+            ('amount_type', '=', 'percent'),
+            ('type_tax_use', '=', invoice.journal_id.type),
+        ])
+        for elem in tree.findall('.//{*}TaxTotal/{*}TaxSubtotal'):
+            percentage = elem.find('.//{*}TaxCategory/{*}Percent')
```

Comment:
You are parsing twice exactly the same data...
At this point, you should already have the total for each account.tax, no need to parse it again.
You should perform the update of tax lines using the same mechanism as the pen on vendor bill